packages/back-end/src/integrations/BigQuery.ts

```diff
@@ -47,7 +48,21 @@ export default class BigQuery extends SqlIntegration {
       useLegacySql: false,
     });
     const [rows] = await job.getQueryResults();
-    return rows;
+    const [metadata] = await job.getMetadata();
```

Comment:
@jdorn , as mentioned on slack, I'm uncertain about this. Before, getting the metadata sometimes would return the "RUNNING" state and no statistics, even though it was below `await job.getQueryResults()`. At least in my test case where I could reliably get the "RUNNING" state, using `await job.getMetadata()` seems to have resolved the issue and it returns the "DONE" metadata instead.

Would like your advice here!

Reply:
If I were to guess, the `job.metadata` property is not meant to be accessed directly.  So using the `getMetadata()` function is likely the proper way to do it.